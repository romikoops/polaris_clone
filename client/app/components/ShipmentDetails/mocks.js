// Will remove it later
// ============================================
import { get } from 'lodash'

export const trucking = { preCarriage: { truckType: '' }, onCarriage: { truckType: '' } }
export const id = 4606
export const selectedDay = '2019-01-24T10:00:00.000Z'

export const ShipmentDetailsAvailableRoutes = [{
  itineraryId: 2863,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'ocean',
  origin: {
    stopId: 4656, hubId: 3023, hubName: 'Gothenburg Port', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4657, hubId: 3027, hubName: 'Qingdao Port', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}, {
  itineraryId: 2958,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'air',
  origin: {
    stopId: 4846, hubId: 3030, hubName: 'Gothenburg Airport', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4847, hubId: 3034, hubName: 'Qingdao Airport', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}]

export const ShipmentDetails = {
  availableRoutes: ShipmentDetailsAvailableRoutes,
  availableMots: ['ocean', 'air']
}

export const cargoItemTypes = [{
  id: 25, dimension_x: null, dimension_y: null, description: 'Pallet', area: null, created_at: '2018-06-27T17:28:28.431Z', updated_at: '2018-06-27T17:28:28.431Z', category: 'Pallet'
}, {
  id: 22, dimension_x: null, dimension_y: null, description: 'Carton', area: null, created_at: '2018-06-27T17:28:28.417Z', updated_at: '2018-06-27T17:28:28.417Z', category: 'Carton'
}, {
  id: 23, dimension_x: null, dimension_y: null, description: 'Crate', area: null, created_at: '2018-06-27T17:28:28.422Z', updated_at: '2018-06-27T17:28:28.422Z', category: 'Crate'
}, {
  id: 26, dimension_x: null, dimension_y: null, description: 'Bottle', area: null, created_at: '2018-06-27T17:28:28.436Z', updated_at: '2018-06-27T17:28:28.436Z', category: 'Bottle'
}, {
  id: 27, dimension_x: null, dimension_y: null, description: 'Stack', area: null, created_at: '2018-06-27T17:28:28.440Z', updated_at: '2018-06-27T17:28:28.440Z', category: 'Stack'
}, {
  id: 28, dimension_x: null, dimension_y: null, description: 'Drum', area: null, created_at: '2018-06-27T17:28:28.445Z', updated_at: '2018-06-27T17:28:28.445Z', category: 'Drum'
}, {
  id: 29, dimension_x: null, dimension_y: null, description: 'Skid', area: null, created_at: '2018-06-27T17:28:28.450Z', updated_at: '2018-06-27T17:28:28.450Z', category: 'Skid'
}, {
  id: 30, dimension_x: null, dimension_y: null, description: 'Barrel', area: null, created_at: '2018-06-27T17:28:28.454Z', updated_at: '2018-06-27T17:28:28.454Z', category: 'Barrel'
}]

export const maxDimensions = {
  general: {
    dimensionX: '590.0', dimensionY: '234.2', dimensionZ: '228.0', payloadInKg: '21770.0', chargeableWeight: '21770.0'
  },
  air: {
    dimensionX: '120.0', dimensionY: '100.0', dimensionZ: '150.0', payloadInKg: '1000.0', chargeableWeight: '1000.0'
  }
}

export const scope = {
  links: { about: '', legal: '' },
  terms: ['You verify that all the information provided above is true', 'You agree to the presented terms and conditions.', 'Demo is to discuss the validity of the presented prices with the product owners.'],
  fee_detail: 'key_and_name',
  closed_shop: false,
  has_customs: true,
  has_insurance: true,
  fixed_currency: false,
  dangerous_goods: false,
  cargo_info_level: 'hs_codes',
  carriage_options: { on_carriage: { export: 'optional', import: 'optional' }, pre_carriage: { export: 'optional', import: 'optional' } },
  detailed_billing: false,
  total_dimensions: true,
  consolidate_cargo: false,
  modes_of_transport: {
    air: { container: true, cargo_item: true }, rail: { container: true, cargo_item: true }, ocean: { container: true, cargo_item: true }, truck: { container: false, cargo_item: false }
  },
  show_beta_features: true,
  closed_registration: false,
  continuous_rounding: false,
  incoterm_info_level: 'text',
  non_stackable_goods: true,
  open_quotation_tool: false,
  customs_export_paper: false,
  fixed_exchange_rates: false,
  require_full_address: true,
  closed_quotation_tool: false
}

export const theme = {
  colors: {
    primary: '#0D5BA9', secondary: '#23802A', brightPrimary: '#2491FD', brightSecondary: '#25ED36'
  },
  logoWide: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_blue.png',
  logoLarge: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoSmall: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoWhite: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_white.png',
  background: 'https://assets.itsmycargo.com/assets/images/cropped_banner_2.jpg'
}

export const origin = {
  latitude: 57.694253, longitude: 11.854048, nexusId: 597, nexusName: 'Gothenburg', country: 'SE'
}
export const destination = {
  latitude: 36.083811, longitude: 120.323534, nexusId: 601, nexusName: 'Qingdao', country: 'CN'
}
export const cargoItem = {
  payloadInKg: 11, totalVolume: 0, totalWeight: 0, dimensionX: 7, dimensionY: 7, dimensionZ: 8, quantity: 7, cargoItemTypeId: 23, dangerousGoods: false, stackable: false
}

export const cargoItemAggregated = {
  payloadInKg: 0, totalVolume: 122, totalWeight: 346, dimensionX: 0, dimensionY: 0, dimensionZ: 0, quantity: 1, cargoItemTypeId: '', dangerousGoods: false, stackable: true
}

export const cargoItemContainer = {
  sizeClass: 'highCube', quantity: 14, dangerousGoods: false, weight: 16
}

export const cargoUnits = [cargoItem]

export const identity = x => x

export const importedProps = {
  ShipmentDetails,
  aggregatedCargo: false,
  cargoItemTypes,
  cargoUnits,
  destination,
  direction: 'export',
  id,
  loadType: 'cargo_item',
  maxDimensions,
  onCarriage: false,
  onChangeCargoUnitCheckbox: identity,
  onChangeCargoUnitInput: identity,
  onChangeCargoUnitSelect: identity,
  onDeleteUnit: identity,
  onUpdateCargoUnit: identity,
  origin,
  preCarriage: false,
  scope,
  selectedDay,
  theme,
  toggleModal: identity,
  trucking
}

export function logPath (props, path) {
  console.log(path, get(props, path, 'NO SUCH PATH'))
}

export function log (props) {
  let holder = '{\n'
  const declarations = []
  Object.keys(props).forEach((key) => {
    const toLog = `const ${key} = ${JSON.stringify(props[key])}`
    if (toLog.length < 150) console.log(toLog)
    declarations.push(toLog)

    holder = `${holder}\n  ${key},`
  })
  holder = `${holder}\n}`
  console.log(declarations.join('\n'))
  // console.log(holder)
  // console.log(JSON.stringify(props, null, 2))
}
