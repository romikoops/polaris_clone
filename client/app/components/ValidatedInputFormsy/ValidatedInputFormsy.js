import React, { Component } from 'react';
import { withFormsy } from 'formsy-react';
import styles from './ValidatedInputFormsy.scss';
import errorStyles from '../../styles/errors.scss';

class ValidatedInputFormsy extends Component {
    constructor(props) {
        super(props);
        this.changeValue = this.changeValue.bind(this);
        this.state = {firstRender: true};
        this.errorsHaveUpdated = false;
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.firstRenderInputs) {
            this.setState({firstRender: true});
        }
    }

    componentDidUpdate() {
        if (this.errorsHaveUpdated) return; // Break the loop if erros have updated

        const event = {target: {name: this.props.name, value: this.props.getValue()}};
        const validationPassed = this.props.isValidValue(event.target.value);

        // break out of function if validation did not pass. This assumes default state in
        // shipmentDetails has errors set to true
        if (!validationPassed) return;

        // Trigger onChange event, and flag errorsHaveUpdated as true, in order to avoid an infinite loop.
        this.props.onChange(event, !validationPassed);
        this.errorsHaveUpdated = true;
    }

    changeValue(event) {
        this.props.onChange(event, !this.props.isValidValue(event.target.value));
        this.props.setFirstRenderInputs(false);
        this.setState({firstRender: false});
        // setValue() will set the value of the component, which in
        // turn will validate it and the rest of the form
        // Important: Don't skip this step. This pattern is required
        // for Formsy to work.
        this.props.setValue(event.currentTarget.value);
    }
    render() {
        // An error message is returned only if the component is invalid
        const errorMessage = this.props.getErrorMessage();
        const inputStyles = {
            width: '100%',
            height: '100%',
            boxSizing: 'border-box'
        };
        const ErrorVisible = this.state.firstRender && !this.props.nextStageAttempt;
        if (!ErrorVisible && !this.props.isValid()) {
            inputStyles.background = 'rgba(232, 114, 88, 0.3)';
            inputStyles.borderColor = 'rgba(232, 114, 88, 0.01)';
            inputStyles.color = 'rgba(211, 104, 80, 1)';
        }
        return (
            <div className={styles.wrapper_input}>
                <input
                	style={inputStyles}
                    onChange={this.changeValue}
                    type={this.props.type}
                    value={(this.props.getValue() || this.props.getValue().toString()) || ''}
                    name={this.props.name}
                />
                <span className={errorStyles.error_message}>{ErrorVisible ? '' : errorMessage}</span>
            </div>
        );
    }
}

export default withFormsy(ValidatedInputFormsy);
