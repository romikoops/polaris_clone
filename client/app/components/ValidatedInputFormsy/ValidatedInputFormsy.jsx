import React, { Component } from 'react'
import { withFormsy } from 'formsy-react'
import PropTypes from 'prop-types'
import styles from './ValidatedInputFormsy.scss'
import errorStyles from '../../styles/errors.scss'

class ValidatedInputFormsy extends Component {
  constructor (props) {
    super(props)
    this.changeValue = this.changeValue.bind(this)
    this.state = { firstRender: true }
    this.errorsHaveUpdated = false
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.firstRenderInputs || nextProps.nextStageAttempt) {
      this.setState({ firstRender: nextProps.firstRenderInputs })
    }
  }

  componentDidUpdate () {
    if (this.errorsHaveUpdated) return // Break the loop if erros have updated
    const event = { target: { name: this.props.name, value: this.props.getValue() } }

    const validationPassed = this.props.isValidValue(event.target.value)

    // break out of function if validation did not pass. This assumes default state in
    // parent has errors set to true
    if (!validationPassed) return

    // Trigger onChange event, and flag errorsHaveUpdated as true,
    // in order to avoid an infinite loop.
    this.props.onChange(event, !validationPassed)
    this.errorsHaveUpdated = true
  }

  changeValue (event) {
    this.props.onChange(event, !this.props.isValidValue(event.target.value))
    if (this.props.setFirstRenderInputs) this.props.setFirstRenderInputs(false)
    this.setState({ firstRender: false })
    // setValue() will set the value of the component, which in
    // turn will validate it and the rest of the form
    // Important: Don't skip this step. This pattern is required
    // for Formsy to work.
    this.props.setValue(event.currentTarget.value)
  }
  render () {
    // An error message is returned only if the component is invalid
    const errorMessage = this.props.getErrorMessage()
    const inputStyles = {
      width: '100%',
      height: '100%',
      boxSizing: 'border-box'
    }
    const ErrorHidden = this.state.firstRender && !this.props.nextStageAttempt
    if (!ErrorHidden && !this.props.isValid()) {
      inputStyles.background = 'rgba(232, 114, 88, 0.3)'
      inputStyles.borderColor = 'rgba(232, 114, 88, 0.01)'
      inputStyles.color = 'rgba(211, 104, 80, 1)'
    }
    // const value = (this.props.getValue() !== undefined &&
    // !Number.isNaN(this.props.getValue())) ? this.props.getValue() : ''
    const value = this.props.getValue()
    return (
      <div className={styles.wrapper_input}>
        <input
          ref={this.props.inputRef}
          style={inputStyles}
          onChange={this.changeValue}
          placeholder={this.props.placeholder}
          type={this.props.type}
          value={value}
          name={this.props.name}
          disabled={this.props.disabled}
          className={this.props.className}
          onKeyDown={this.props.onKeyDown}
          min={this.props.min}
        />
        <span
          className={errorStyles.error_message}
          style={this.props.errorStyles}
        >
          {ErrorHidden ? '' : errorMessage}
        </span>
      </div>
    )
  }
}

ValidatedInputFormsy.propTypes = {
  isValidValue: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  inputRef: PropTypes.element,
  isValid: PropTypes.func.isRequired,
  getErrorMessage: PropTypes.func.isRequired,
  setValue: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  getValue: PropTypes.func.isRequired,
  setFirstRenderInputs: PropTypes.func,
  firstRenderInputs: PropTypes.bool,
  nextStageAttempt: PropTypes.bool,
  onKeyDown: PropTypes.func,
  min: PropTypes.string,
  errorStyles: PropTypes.objectOf(PropTypes.string)
}

ValidatedInputFormsy.defaultProps = {
  disabled: false,
  placeholder: '',
  setFirstRenderInputs: null,
  firstRenderInputs: false,
  nextStageAttempt: false,
  onKeyDown: null,
  min: '',
  errorStyles: {},
  inputRef: null,
  className: ''
}

export default withFormsy(ValidatedInputFormsy)
