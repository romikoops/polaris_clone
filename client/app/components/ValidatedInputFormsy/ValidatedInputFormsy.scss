:local {
	.wrapper_input {
		width: 100%;
		height: 100%;
		position: relative;
	}
	span.error_message {
		position: absolute;
		left: 0;
		bottom: -25px;
		color: rgba(black, 0.5);
	}
}