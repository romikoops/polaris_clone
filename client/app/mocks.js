import { set, cloneDeep } from 'lodash'

/**
 * Checks if variable is an non-empty object
 *
 * True for `{a:1}`
 * Fase for `null, {}, [], [1,2,3]`
 */
const isObject = (x) => {
  const ok = x !== null && !Array.isArray(x) && typeof x === 'object'
  if (!ok) {
    return false
  }

  return Object.keys(x).length > 0
}
/* eslint-disable */
/**
 * Used in unit test to modify specific properties
 * with minimal lines of code
 *
 * You can specify `path` to be empty string
 * if the change affects more than one of object's branches
 */
export const change = (origin, pathRaw, rules) => {
  const willReturn = cloneDeep(origin)
  
  if (!isObject(rules)) {
    set(willReturn, pathRaw, rules)
    return willReturn
  }
  const path = pathRaw === '' ? '' : `${pathRaw}.`
  for (const ruleKey of Object.keys(rules)) {
    const rule = rules[ruleKey]
    if (!isObject(rule)) {
      set(
        willReturn,
        `${path}${ruleKey}`,
        rule
      )
      continue
    }
    Object.keys(rule).filter(subruleKey => !isObject(rule[subruleKey])).map((subruleKey) => {
      const subrule = rule[subruleKey]
      set(
        willReturn,
        `${path}${ruleKey}.${subruleKey}`,
        subrule
      )
    })
    Object.keys(rule).filter(subruleKey => isObject(rule[subruleKey])).map((subruleKey) => {
      const subrule = rule[subruleKey]
      Object.keys(subrule).map((deepKey) => {
        const deep = rule[subruleKey][deepKey]
        set(
          willReturn,
          `${path}${ruleKey}.${subruleKey}.${deepKey}`,
          deep
        )
      })
    })
  }

  return willReturn
}
/* eslint-enable */
export const identity = input => input

export const theme = {
  colors: {
    primary: '#333',
    secondary: '#fafafa'
  }
}

export const user = {
  role: { name: 'shipper' },
  company_name: 'FOO_COMPANY',
  currency: 'EUR',
  email: 'foo@bar.baz',
  first_name: 'John',
  guest: false,
  last_name: 'Doe',
  phone: '6345789'
}

export const history = {
  push: identity
}

export const tenant = {
  data: {
    id: 123,
    scope: {
      modes_of_transport: {
        ocean: {
          OCEAN_LOAD_TYPE: true
        },
        air: {},
        truck: {},
        rail: {}
      }
    },
    theme,
    subdomain: 'foosubdomain'
  }
}

export const req = {
  schedule: [{}],
  total: 345,
  shipment: {}
}

const schedulesInShipmentData = [
  { hub_route_key: 'OCEAN_HUB_ROUTE_KEY', mode_of_transport: 'ocean' },
  { hub_route_key: 'AIR_HUB_ROUTE_KEY', mode_of_transport: 'air' },
  { hub_route_key: 'TRUCK_HUB_ROUTE_KEY', mode_of_transport: 'truck' },
  { hub_route_key: 'RAIL_HUB_ROUTE_KEY', mode_of_transport: 'rail' }
]

export const shipmentInShipmentData = {
  load_type: 'OCEAN_LOAD_TYPE',
  selected_offer: { cargo: {}, total: { value: 77 } },
  schedules_charges: {
    OCEAN_HUB_ROUTE_KEY: { total: 40 },
    AIR_HUB_ROUTE_KEY: { total: 200 },
    TRUCK_HUB_ROUTE_KEY: { total: 75 },
    RAIL_HUB_ROUTE_KEY: { total: 125 }
  },
  total_goods_value: {
    value: 15,
    currency: 'USD'
  },
  trucking: {
    pre_carriage: { trucking_time_in_seconds: 55 }
  },
  has_on_carriage: false,
  has_pre_carriage: false,
  notes: 'FOO_NOTES',
  cargo_notes: 'FOO_CARGO_NOTES',
  eori: 1234,
  incoterm_text: 'FOO_INCOTERM_TEXT',
  total_price: {
    value: 12,
    currency: 'USD'
  }
}

export const shipmentData = {
  contacts: [],
  shipment: shipmentInShipmentData,
  documents: [],
  cargoItems: [],
  containers: [],
  schedules: schedulesInShipmentData,
  locations: {
    startHub: 'FOO_START_HUB',
    endHub: 'FOO_END_HUB',
    destination: {
      street_number: 9,
      street: 'BAR_STREET',
      city: 'BAR_CITY',
      country: 'China',
      zip_code: 845321
    },
    origin: {
      street_number: 7,
      street: 'FOO_STREET',
      city: 'FOO_CITY',
      country: 'Germany',
      zip_code: 21177
    }
  }
}

export const match = {
  params: {}
}

export const shipments = {
  direction: 'export',
  finished: [],
  open: [],
  requested: []
}

export const shipment = {
  origin_hub: { name: 'FOO_ORIGIN_HUB' },
  destination_hub: { name: 'FOO_DESTINATION_HUB' },
  id: 654,
  status: 'FOO_STATUS',
  clientName: 'FOO_CLIENT_NAME',
  planned_etd: 789,
  imc_reference: 'FOO_IMC_REFERENCE',
  schedule_set: []
}

export const location = {
  primary: true,
  id: 246,
  street_number: 579,
  streetNumber: 579,
  geocodedAddress: 'FOO_GEOCODED_ADDRESS',
  street: 'FOO_STREET',
  city: 'FOO_CITY',
  zip_code: '22456',
  zipCode: '22456',
  country: 'Germany'
}

export const address = {
  id: 13579,
  street_number: '1',
  street: 'Uhlandweg',
  city: 'Hamburg',
  zip_code: '22848',
  country: 'Germany'
}

export const hub = {
  location,
  name: 'FOO_HUB_NAME'
}

export const client = {
  first_name: 'Sebastian',
  last_name: 'Muller',
  email: 'muller55@yahoo.de',
  company_name: 'Mehristweniger',
  password: 'pwd'
}

export const route = {
  id: 24680
}

export const vehicleType = {
  id: 827364,
  name: 'FOO_VEHICLE_TYPE_NAME',
  mode_of_transport: 'FOO_VEHICLE_TYPE_TRANSPORT'
}

export const schedule = {
  hub_route_key: 'FOO_SCHEDULE_ROUTE_KEY',
  id: 555777,
  eta: 441188
}

export const charge = {
  id: 654297
}

export const locations = {
  origin: {},
  destination: {}
}

class MapMock {
  constructor (x) {
    this.x = x
  }
  bindTo () {
    return this.x
  }
  setContent () {
    return this.x
  }
  addListener () {
    return this.x
  }
}

export const gMaps = {
  InfoWindow: MapMock,
  LatLngBounds: MapMock,
  Map: MapMock,
  MapTypeId: { ROADMAP: '' },
  Marker: MapMock,
  Point: MapMock,
  Size: MapMock,
  places: { Autocomplete: MapMock }
}
