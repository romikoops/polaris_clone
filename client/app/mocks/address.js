export const firstAddress = {
  user: {
    primary: true
  },
  primary: true,
  id: 246,
  streetNumber: 'FOO_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'FOO_GEOCODED_ADDRESS',
  street: 'FOO_ADDRESS_STREET',
  city: 'FOO_ADDRESS_CITY',
  zipCode: 'FOO_ADDRESS_ZIP_CODE',
  country: 'FOO_ADDRESS_COUNTRY'
}

export const secondAddress = {
  user: {
    primary: false
  },
  primary: false,
  id: 704,
  streetNumber: 'BAR_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'BAR_GEOCODED_ADDRESS',
  street: 'BAR_ADDRESS_STREET',
  city: 'BAR_ADDRESS_CITY',
  zipCode: 'BAR_ADDRESS_ZIP_CODE',
  country: 'BAR_ADDRESS_COUNTRY'
}

export const originAddress = {
  city: 'ORIGIN_ADDRESS_CITY',
  country: 'ORIGIN_ADDRESS_COUNTRY',
  street: 'ORIGIN_ADDRESS_STREET',
  street_number: 'ORIGIN_ADDRESS_STREET_NUMBER',
  zip_code: 'ORIGIN_ADDRESS_ZIP_CODE'
}
export const destinationAddress = {
  city: 'DESTINATION_ADDRESS_CITY',
  country: 'DESTINATION_ADDRESS_COUNTRY',
  street: 'DESTINATION_ADDRESS_STREET',
  street_number: 'DESTINATION_ADDRESS_STREET_NUMBER',
  zip_code: 'DESTINATION_ADDRESS_ZIP_CODE'
}

export const addresses = {
  origin: originAddress,
  destination: destinationAddress
}
