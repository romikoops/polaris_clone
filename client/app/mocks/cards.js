export const firstCard = {
  name: 'FOO_NAME',
  img: 'FOO_IMG',
  url: 'FOO_URL',
  handleClick: x => x,
  options: {
    name: 'FOO_OPTIONS_NAME'
  }
}
export const secondCard = {
  name: 'BAR_NAME',
  img: 'BAR_IMG',
  url: 'BAR_URL',
  handleClick: x => x,
  options: {
    name: 'BAR_OPTIONS_NAME'
  }
}
export const cards = [
  firstCard,
  secondCard
]
