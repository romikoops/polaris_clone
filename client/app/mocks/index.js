export * from './address'
export * from './cards'
export * from './cargoItems'
export * from './client'
export * from './consignee'
export * from './contacts'
export * from './containers'
export * from './currencies'
export * from './dimensions'
export * from './documents'
export * from './feeHash'
export * from './functions'
export * from './gMaps'
export * from './group'
export * from './hub'
export * from './match'
export * from './notifyees'
export * from './offer'
export * from './results'
export * from './routes'
export * from './schedules'
export * from './scope'
export * from './shipmentData'
export * from './shipments'
export * from './shipper'
export * from './tenant'
export * from './theme'
export * from './user'
