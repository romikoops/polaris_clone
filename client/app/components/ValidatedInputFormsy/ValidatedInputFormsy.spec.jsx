import * as React from 'react'
import { shallow } from 'enzyme'
import { identity } from '../../mocks/index'

import ValidatedInputFormsy from './ValidatedInputFormsy'

// Without the mock, we get the following error
// TypeError: Cannot read property 'attachToForm' of undefined
// ============================================
jest.mock('formsy-react', () => ({
  withFormsy: x => x
}))

const propsBase = {
  isValidValue: identity,
  name: 'NAME',
  className: 'CLASSNAME',
  disabled: false,
  type: 'TYPE',
  placeholder: 'PLACEHOLDER',
  inputRef: <div />,
  isValid: identity,
  getErrorMessage: identity,
  setValue: identity,
  onChange: identity,
  getValue: identity,
  setFirstRenderInputs: identity,
  firstRenderInputs: false,
  nextStageAttempt: false,
  onKeyDown: identity,
  min: 'MIN',
  errorStyles: {}
}

test('shallow render', () => {
  expect(shallow(<ValidatedInputFormsy {...propsBase} />)).toMatchSnapshot()
})

test('props.nextStageAttempt is true', () => {
  const props = {
    ...propsBase,
    nextStageAttempt: true
  }
  expect(shallow(<ValidatedInputFormsy {...props} />)).toMatchSnapshot()
})
