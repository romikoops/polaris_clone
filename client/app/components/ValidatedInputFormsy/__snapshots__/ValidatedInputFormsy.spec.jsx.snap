// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props.nextStageAttempt is true 1`] = `
<div
  className="wrapper_input"
>
  <input
    className="FOO_CLASSNAME"
    data-hj-whitelist={true}
    disabled={false}
    min="FOO_MIN"
    name="FOO_NAME"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="FOO_PLACEHOLDER"
    style={
      Object {
        "background": "rgba(232, 114, 88, 0.3)",
        "borderColor": "rgba(232, 114, 88, 0.01)",
        "boxSizing": "border-box",
        "color": "rgba(211, 104, 80, 1)",
        "height": "100%",
        "width": "100%",
      }
    }
    type="FOO_TYPE"
  />
  <span
    className="error_message"
    style={Object {}}
  />
</div>
`;

exports[`shallow render 1`] = `
<div
  className="wrapper_input"
>
  <input
    className="FOO_CLASSNAME"
    data-hj-whitelist={true}
    disabled={false}
    min="FOO_MIN"
    name="FOO_NAME"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="FOO_PLACEHOLDER"
    style={
      Object {
        "boxSizing": "border-box",
        "height": "100%",
        "width": "100%",
      }
    }
    type="FOO_TYPE"
  />
  <span
    className="error_message"
    style={Object {}}
  />
</div>
`;
